src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentsService.kt

```diff
@@ -85,11 +91,34 @@ class AppointmentsService(
     notifyProbationPractitioner: Boolean,
   ): ActionPlanAppointment {
     val appointment = getActionPlanAppointmentOrThrowException(actionPlanId, sessionNumber)
+
+    if (appointment.sessionFeedbackSubmittedAt != null) {
+      throw ResponseStatusException(HttpStatus.CONFLICT, "session feedback has already been submitted for this appointment")
+    }
+
     setBehaviourFields(appointment, behaviourDescription, notifyProbationPractitioner)
-    appointmentEventPublisher.behaviourRecordedEvent(appointment, notifyProbationPractitioner)
     return actionPlanAppointmentRepository.save(appointment)
   }
 
+  fun submitSessionFeedback(actionPlanId: UUID, sessionNumber: Int): ActionPlanAppointment {
+    val appointment = getActionPlanAppointmentOrThrowException(actionPlanId, sessionNumber)
+
+    if (appointment.sessionFeedbackSubmittedAt != null) {
+      throw ResponseStatusException(HttpStatus.CONFLICT, "session feedback has already been submitted for this appointment")
+    }
+
+    if (appointment.attendanceBehaviourSubmittedAt == null || appointment.attendanceSubmittedAt == null) {
+      throw ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "can't submit session feedback unless attendance and behaviour have been recorded")
```

Comment:
Is this a good time to look at how to surface this in the API documentation?

Reply:
yeh good idea.